applications/processor/twitter-trend-processor/src/test/java/org/springframework/cloud/stream/app/processor/twitter/trend/TwitterTrendLocationProcessorIntegrationTests.java

```diff
@@ -0,0 +1,226 @@
+						"--twitter.connection.rawJson=true",
+
+						"--twitter.connection.consumerKey=myConsumerKey",
+						"--twitter.connection.consumerSecret=myConsumerSecret",
+						"--twitter.connection.accessToken=myAccessToken",
+						"--twitter.connection.accessTokenSecret=myAccessTokenSecret")) {
+
+			InputDestination input = context.getBean(InputDestination.class);
+			OutputDestination output = context.getBean(OutputDestination.class);
+			assertThat(input).isNotNull();
+			assertThat(output).isNotNull();
+
+			input.send(new GenericMessage<>("hello".getBytes(StandardCharsets.UTF_8)));
+
+			Message<byte[]> outputMessage = output.receive(Duration.ofSeconds(300).toMillis());
+			assertThat(outputMessage).isNotNull();
+
+			mockClient.verify(availableTrendsRequest, once());
+			assertThat(outputMessage).isNotNull();
+
+			String payload = new String(outputMessage.getPayload());
+
+			assertThat(payload).contains("placeType");
+			assertThat(payload).doesNotContain("placeCode");
+			assertThat(payload).doesNotContain("countryName");
+		}
+	}
+
+	@Test
+	public void testTwitterClosestTrends() {
+		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(
+				TestChannelBinderConfiguration.getCompleteConfiguration(TwitterTrendProcessorIntegrationTests.TestTwitterTrendProcessorApplication.class))
+				.web(WebApplicationType.NONE)
+				.run("--spring.cloud.stream.function.definition=trendOrTrendLocationsFunction",
+
+						"--twitter.trend.trendQueryType=trendLocation",
+						"--twitter.connection.rawJson=true",
+						"--twitter.trend.closest.lat='52.379189'",
+						"--twitter.trend.closest.lon='4.899431'",
+
+						"--twitter.connection.consumerKey=myConsumerKey",
+						"--twitter.connection.consumerSecret=myConsumerSecret",
+						"--twitter.connection.accessToken=myAccessToken",
+						"--twitter.connection.accessTokenSecret=myAccessTokenSecret")) {
+
+			InputDestination input = context.getBean(InputDestination.class);
+			OutputDestination output = context.getBean(OutputDestination.class);
+			assertThat(input).isNotNull();
+			assertThat(output).isNotNull();
+
+			input.send(new GenericMessage<>("hello".getBytes(StandardCharsets.UTF_8)));
+
+			Message<byte[]> outputMessage = output.receive(Duration.ofSeconds(300).toMillis());
+			assertThat(outputMessage).isNotNull();
+
+			mockClient.verify(closestTrendsRequest, once());
+			assertThat(outputMessage).isNotNull();
+		}
+	}
+
+	public static HttpRequest setExpectation(HttpRequest request) {
+		mockClient
+				.when(request, exactly(1))
+				.respond(response()
+						.withStatusCode(200)
+						.withHeaders(
+								new Header("Content-Type", "application/json; charset=utf-8"),
+								new Header("Cache-Control", "public, max-age=86400"))
+						.withBody(TwitterTestUtils.asString("classpath:/response/trend_locations.json"))
+						.withDelay(TimeUnit.SECONDS, 1)
+				);
+		return request;
+	}
+
+	@SpringBootConfiguration
+	@EnableAutoConfiguration
```

Comment:
Because of the `ComponentScan` (i guess) coming with `@SpringBootApplication` you can not have multiple integration tests (e.g. multiple tests with @SpringBootApplication) at the same package level.  Beens got overwritten ... I believe you can do it if carefully separate the tests in inner packages or define common test spring boot app class.